core/executor/benches/revm_adapter.rs

```diff
@@ -0,0 +1,335 @@
+                _ => Account {
+                    nonce:        U256::zero(),
+                    balance:      U256::zero(),
+                    storage_root: RLP_NULL,
+                    code_hash:    NIL_DATA,
+                },
+            };
+
+            let storage_root = old_account.storage_root;
+            let mut storage_trie = if storage_root == RLP_NULL {
+                MPTTrie::new(Arc::clone(&self.db))
+            } else {
+                MPTTrie::from_root(storage_root, Arc::clone(&self.db)).unwrap()
+            };
+            change.storage.into_iter().for_each(|(k, v)| {
+                let _ = storage_trie.insert(u256_to_u8_slice(&k), u256_to_u8_slice(&v));
+            });
+
+            let code_hash = if let Some(code) = change.info.code {
+                let code_hash = change.info.code_hash;
+                if code_hash != old_account.code_hash {
+                    blocking_async!(
+                        self,
+                        storage,
+                        insert_code,
+                        Context::new(),
+                        addr.into(),
+                        code_hash,
+                        code.bytes().clone()
+                    );
+                }
+                code_hash
+            } else {
+                NIL_DATA
+            };
+
+            let new_account = Account {
+                nonce: U256::from(change.info.nonce),
+                balance: change.info.balance,
+                storage_root: storage_trie.commit().unwrap(),
+                code_hash,
+            };
+
+            let account_bytes = new_account.encode().unwrap();
+            self.trie
+                .insert(addr.as_bytes(), account_bytes.as_ref())
+                .unwrap();
+        });
+    }
+}
+
+impl<S, DB> RevmAdapter<S, DB>
+where
+    S: Storage + 'static,
+    DB: cita_trie::DB + 'static,
+{
+    pub fn init_mpt(&mut self, account: Account, addr: Address) {
+        let mut mpt = MPTTrie::new(Arc::clone(&self.db));
+        let distribute_account = account;
+
+        mpt.insert(
+            addr.as_slice(),
+            distribute_account.encode().unwrap().as_ref(),
+        )
+        .unwrap();
+
+        mpt.commit().unwrap();
+        self.trie = mpt;
+    }
+
+    pub fn new(storage: S, db: DB, exec_ctx: ExecutorContext) -> Self {
+        let db = Arc::new(db);
+        RevmAdapter {
+            exec_ctx,
+            trie: MPTTrie::new(Arc::clone(&db)),
+            storage: Arc::new(storage),
+            db,
+        }
+    }
+}
+
+#[inline]
+fn u256_to_u8_slice(index: &U256) -> &[u8] {
+    let u64_slice = index.as_ref();
+    let result: &[u8] = bytemuck::cast_slice(u64_slice);
+    result
+}
+
+#[inline]
+fn u8_slice_to_u256(bytes: &[u8]) -> Option<U256> {
+    let u64_slice: &[u64] = bytemuck::cast_slice(bytes);
+    let array: &[u64; 4] = u64_slice.try_into().expect("incorrect length");
+    Some(U256(array.to_owned()))
+}
```

Comment:
Why not use `U256::to_le_bytes(&self)` and `U256::from_le_bytes` method.

Reply:
I have tried writing such like this:
```rust
    use protocol::types::U256;

    fn u256_to_u8_slice_2(index: U256) -> Vec<u8> {
        let mut bytes = vec![];
        index.to_little_endian(&mut bytes);
        bytes
    }

    fn u8_slice_to_u256_2(bytes: &[u8]) -> U256 {
        U256::from_little_endian(bytes)
    }

    #[test]
    fn convert_test() {
        let num = U256::from(12345678910099u64);
        let slice = u256_to_u8_slice_2(num);
        let actual = u8_slice_to_u256_2(&slice);
        assert_eq!(num, actual);
    }
```

And it went wrong.

Output:
```
running 1 test
thread 'system::tests::convert_test' panicked at 'assertion failed: 4 * 8 == bytes.len()', /Users/jeremyhe/.cargo/registry/src/github.com-1ecc6299db9ec823/primitive-types-0.11.1/src/lib.rs:38:1
```

Reply:
The `to_little_endian` require the given vector with the 32 length. You  can write like this:
```
let mut bytes = vec![0u8; 32];
index.to_little_endian(&mut bytes); 
bytes
```

Reply:
OK.
I wrote a benchmark for these 2 methods:
```rust
use criterion::{criterion_group, criterion_main, Criterion};

use protocol::types::U256;

fn u256_to_u8_slice(index: &U256) -> &[u8] {
    let u64_slice = index.as_ref();
    let result: &[u8] = bytemuck::cast_slice(u64_slice);
    result
}

fn u256_to_u8_slice_2(index: U256) -> Vec<u8> {
    let mut bytes = vec![0u8; 32];
    index.to_little_endian(&mut bytes);
    bytes
}

fn u8_slice_to_u256_2(bytes: &[u8]) -> U256 {
    U256::from_little_endian(bytes)
}

fn u8_slice_to_u256(bytes: &[u8]) -> Option<U256> {
    let u64_slice: &[u64] = bytemuck::cast_slice(bytes);
    let array: &[u64; 4] = u64_slice.try_into().expect("incorrect length");
    Some(U256(array.to_owned()))
}

fn criterion_convert(c: &mut Criterion) {
    let num = U256::from(12345678910099u64);
    c.bench_function("convert1", |b| {
        b.iter(|| {
            let slice = u256_to_u8_slice(&num);
            let actual = u8_slice_to_u256(slice).unwrap();
            assert_eq!(num, actual);
        });
    });
    c.bench_function("convert2", |b| {
        b.iter(|| {
            let slice = u256_to_u8_slice_2(num);
            let actual = u8_slice_to_u256_2(&slice);
            assert_eq!(num, actual);
        });
    });
}

criterion_group!(benches, criterion_convert);
criterion_main!(benches);

```

And the result:
- convert1: 858.84ps
- convert2: 32.415ns

If this benchmark should be put into this PR?

Reply:
It is up to you.